### ISSUE TITLE
Travis CI is becoming useless

### ISSUE THREAD

**Issue by charris** [numpy/numpy#17656]
Travis CI has become too slow, simple PRs can take overnight to finish. Some of that is probably extra testing on the part of NumPy, OTOH, I look at the stats and active opensource builds are [capped at 550 with a backlog of 10K](https://www.traviscistatus.com/) That doesn't seem healthy.

**Comment by WarrenWeckesser** [comment]
Another data point: lately, it can take more than 10 hours for Travis CI to start on scipy pull requests.

**Comment by jamesmyatt** [comment]
I bet they're overloaded by hacktoberfest

**Comment by melissawm** [comment]
I have seen people from other projects with similar issues, and it [may be related](https://travis-ci.community/t/builds-hang-in-queued-state/10250/6) to the planned move from .org to .com

**Comment by andyfaff** [comment]
GitHub actions are easy to setup, with good performance. Perhaps numpy could spread the load out over different ci providers. I think Travis is the only one that's good enough for arm64 and ppc though.

**Comment by hugovk** [comment]
Some more on the state of Travis: https://github.com/psf/black/pull/1790#issuecomment-717545031

It's probably worth ditching Travis for GitHub Actions except for the unique CPU architectures like arm64, ppc, s390x.

**Comment by mattip** [comment]
> it may be related to the planned move from .org to .com

That might be optimistic. Don't they both use the same build infrastructure? The status page does not distinguish between them. On the other hand, looking at the status by month, it seems the spike started around Oct 22, In previous spikes, the number of "Active Linux Builds" changed in accordance to the load, now it seems to be pinned to ~550.

**Comment by hugovk** [comment]
Travis CI staff say as much [in the link](https://travis-ci.community/t/builds-hang-in-queued-state/10250/6):

> Unfortunately, this might happen as we move infrastructure over from .org to .com 

And repos I've moved over have gone from sluggish, <5 jobs at .org to fast, 7 jobs(!) at .com. But history is not migrated at the moment and it's not clear if it will be:

* https://travis-ci.community/t/will-builds-logs-be-migrated-upon-org-com-migration/7906

**Comment by mattip** [comment]
The other downside to moving to travis-ci.com is that to log in, you need to give all the permissions for all your repos, public and private. This could be a blocker for some admins who wish to restart or control jobs.

**Comment by charris** [comment]
> And repos I've moved over have gone from sluggish, <5 jobs at .org to fast

How exactly is the move made? My impression is that someone needs to register with travis-ci.com and add the numpy repo(s), but it wasn't clear to me if the current webhooks needed to be changed or if there was an app.

**Comment by hugovk** [comment]
There's instructions here:

* https://docs.travis-ci.com/user/migrate/open-source-repository-migration#migrating-a-repository

Once the initial step is done to setup the app, there's a button to click per repo.

I didn't need to update any webhooks, although one repo did need to update protected branch check for using automerge (the check name is a bit different), see:

* https://docs.travis-ci.com/user/migrate/open-source-repository-migration#do-i-need-to-make-any-other-changes.

I've only done stuff from my own account so far, not any orgs.

**Comment by mattip** [comment]
Someone who is the admin of the numpy organization needs to do it.

**Comment by charris** [comment]
I don't see that we have much choice about making the move, current travis-ci can't do the job. Should I go ahead and give it a shot, or should we discuss it this afternoon during the triage meeting?

**Comment by mattip** [comment]
The transition will need to be done anyway by Dec 31, so I vote for going ahead now.

**Comment by charris** [comment]
I've given it a first try, don't know if it worked :) There is now a `travis-ci` entry in `Installed Github Apps`

**Comment by charris** [comment]
But it isn't working. I swear, the migration documentation manages to be completely useless without being illiterate. It is a marvel in its own right. I don't even know if there is an org login, all I can do is login personally, and then I can add every numpy repo except numpy itself.

**Comment by seberg** [comment]
Oh, and I logged in and thought I did not see the numpy main repo because of insufficient rights for some reason, rather than a general thing.  Maybe we have to disable on the `.org` domain first?!

**Comment by mattip** [comment]
Are there different permissions on the numpy/numpy repo from the rest? You need to be an admin both on the numpy org and then numpy/numpy repo, then on your travis-ci.com dashboard refresh the github connection (on the left).

**Comment by mattip** [comment]
sorry owner, not admin

**Comment by charris** [comment]
I am an owner of the organization.

**Comment by charris** [comment]
Well, strictly speaking, we no longer have owners, but I am an org-admin.

**Comment by charris** [comment]
OK, I think it is working. I reinstalled the app -- the previous version was very old-- then had to sign up the numpy org for the beta, and then was able to migrate four repos. TBH, I cannot remember all the exact steps I went through, I wasn't born to do this sort of work :) Also disabled the previous webhook that went to travis-ci.org, doesn't seem to be needed anymore, probably would lead to duplicate tests.
